### PR TITLE
Add some more weird-exprs

### PR DESCRIPTION
**Pull request by oberien**
Add or-pattern syntax in argument position to `special_characters` in `weird-exprs.rs`.

### CODE REVIEW

**Comment by joshtriplett** [comment]
Please don't drop the emoji in the single quotes; that's a test in itself.

**Comment by oberien** [comment]
I added a different function which contains the emoji. It appears as though the github (and play.rust-lang.org) syntax highlighters don't highlight `fn` correctly if followed by a unicode identifier.

I also added a function returning itself such that it can be curried infinitely many times.

**Comment by the8472** [comment]
>  It appears as though the github (and play.rust-lang.org) syntax highlighters don't highlight `fn` correctly if followed by a unicode identifier.

Instead of changing the code due to inaccurate code highlighters, wouldn't it be better to file a bug against the highlighters?

**Comment by joshtriplett** [comment]
@the8472 Agreed. The purpose of these expressions is to invoke weird corner cases. Sounds like they found one, in syntax highlighters. That's all the more reason to *keep* those expressions; they might cause similar bugs in other things.

I would propose that we keep the expression as-is, and add a *new* weird expression that includes additional components (potentially copying from the existing one if there's an interesting interaction).

**Comment by oberien** [comment]
Oh yeah, I absolutely won't change the code. It is completely valid and working rust code. The syntax highlighting was just something I noticed while working on it. The whole file is just weird expressions to trigger exactly these kinds of problems that most likely won't be found in "normal" crates or projects.

> I would propose that we keep the expression as-is, and add a new weird expression that includes additional components (potentially copying from the existing one if there's an interesting interaction).

I'm not sure I understand what you mean. The code in this PR should be finished from my side.

File: src/test/ui/weird-exprs.rs [hunk 1]
```diff
@@ -116,7 +118,7 @@ fn union() {
 }
 
 fn special_characters() {
-    let val = !((|(..):(_,_),__@_|__)((&*"\\",'🤔')/**/,{})=={&[..=..][..];})//
+    let val = !((|(..):(_,_),(|__@_|__)|__)((&*"\\",'#')/**/,{})=={&[..=..][..];})//
```
Comment (the8472) [inline src/test/ui/weird-exprs.rs:121]: > > I would propose that we keep the expression as-is, and add a new weird expression that includes additional components (potentially copying from the existing one if there's an interesting interaction).
> 
> I'm not sure I understand what you mean. The code in this PR should be finished from my side.

**Comment by crlf0710** [comment]
@oberien Ping from triage, any updates here?

**Comment by JohnCSimon** [comment]
Ping from triage: 
@oberien I'm closing this as inactive, please feel free to reopen when you're ready to continue. Thank you.

**Pull request closed without merging**